loopd/swapclient_server.go

```diff
@@ -217,20 +217,36 @@ func (s *swapClientServer) marshallSwap(loopSwap *loop.SwapInfo) (
 	}
 
 	var swapType clientrpc.SwapType
-	var htlcAddress, htlcAddressP2WSH, htlcAddressNP2WSH string
+	var (
+		htlcAddress       string
+		htlcAddressP2TR   string
+		htlcAddressP2WSH  string
+		htlcAddressNP2WSH string
+	)
 	var outGoingChanSet []uint64
 	var lastHop []byte
 
 	switch loopSwap.SwapType {
 	case swap.TypeIn:
 		swapType = clientrpc.SwapType_LOOP_IN
-		htlcAddressP2WSH = loopSwap.HtlcAddressP2WSH.EncodeAddress()
+		scriptVersion := loop.GetHtlcScriptVersion(
```

Comment:
One thing I've noticed here is the version info is implicitly encoded in the htlc address already. By checking `loopSwap.HtlcAddressP2TR` against nil we'd know the current version. This could clear the logical branches a bit, here and a few other places. However it's not as explicit, so no strong opinions here.